File: kill-port.js
const { execSync } = require('child_process');

function killPort(port) {
  try {
    console.log(`Finding process on port ${port}...`);
    
    // Find process
    const result = execSync(`netstat -ano | findstr :${port}`, { encoding: 'utf8' });
    const lines = result.trim().split('\n');
    const pids = new Set();
    
    lines.forEach(line => {
      const parts = line.trim().split(/\s+/);
      if (line.includes('LISTENING') && parts.length >= 5) {
        const pid = parts[parts.length - 1];
        if (/^\d+$/.test(pid)) {
          pids.add(pid);
        }
      }
    });
    
    if (pids.size === 0) {
      console.log(`No process found on port ${port}`);
      return;
    }
    
    // Kill processes
    pids.forEach(pid => {
      console.log(`Killing process ${pid}...`);
      try {
        execSync(`taskkill /F /PID ${pid}`, { stdio: 'inherit' });
      } catch (e) {
        console.log(`Failed to kill ${pid}`);
      }
    });
    
    console.log(`Successfully killed ${pids.size} process(es)`);
    
  } catch (error) {
    if (error.message.includes('command failed')) {
      console.log(`No process found on port ${port}`);
    } else {
      console.error('Error:', error.message);
    }
  }
}

killPort(3002);

File: kill_port.py
import os
import subprocess
import sys

def kill_process_on_port(port):
    try:
        # Find process using the port
        result = subprocess.run(
            f'netstat -ano | findstr :{port}',
            shell=True,
            capture_output=True,
            text=True
        )
        
        if result.returncode != 0:
            print(f"No process found on port {port}")
            return
        
        lines = result.stdout.strip().split('\n')
        pids = set()
        
        for line in lines:
            parts = line.split()
            if len(parts) >= 5 and 'LISTENING' in line:
                pid = parts[-1]
                if pid.isdigit():
                    pids.add(pid)
        
        if not pids:
            print(f"No LISTENING process found on port {port}")
            return
        
        # Kill each process
        for pid in pids:
            print(f"Killing process {pid}...")
            subprocess.run(f'taskkill /F /PID {pid}', shell=True)
        
        print(f"Successfully killed {len(pids)} process(es)")
        
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    kill_process_on_port(3002)

